### ISSUE TITLE
Query on recreation issue - Macro F1 or weighted F1 usesd? How skewness is handled?

### ISSUE THREAD

**Issue by PraAnj** [zcakhaa/DeepLOB-Deep-Convolutional-Neural-Networks-for-Limit-Order-Books#14]
I am trying to recreate your scores on F1. Can you please kindly provide insights into how the F1 is calculated on F1-2010 dataset scores? You got 83.4% on k=10 as the F1. 

01) Is the above F1 value taken by simple averaging (macro average) between per class F1 values? or by weighted averaging (weighted average) of per class F1 values?
02) k=10 is somewhat 1:3:1 like class distribution. Is your model performs equally better on all classes or performs better for stationary class more?
03) Did you do any class weighting for training on k=10? If not can I know the characteristics of your model which handles the class skewness?

**Comment by zcakhaa** [comment]
1.  It should be the weighted avg from sklearn.classification_report if I am not wrong. It has been a while. You can also check this one

https://github.com/zcakhaa/Multi-Horizon-Forecasting-for-Limit-Order-Books/blob/main/code_gpu/run_train_deeplob_attention.ipynb

  1.  It performs well in all classes. You can check the confusion matrices.
  2.  I don't do class weighting or weight calibrated loss function, but it would be worthing looking into it. Btw, FI-2010 is relatively easy to predict, so I suggest you take this dataset as a demonstration.


________________________________
From: Prageeth Anjula ***@***.***>
Sent: Sunday, June 6, 2021 8:29 AM
To: zcakhaa/DeepLOB-Deep-Convolutional-Neural-Networks-for-Limit-Order-Books ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [zcakhaa/DeepLOB-Deep-Convolutional-Neural-Networks-for-Limit-Order-Books] Query on recreation issue - Macro F1 or weighted F1 usesd? How skewness is handled? (#14)


I am trying to recreate your scores on F1. Can you please kindly provide insights into how the F1 is calculated on F1-2010 dataset scores? You got 83.4% on k=10 as the F1.

  1.  Is the above F1 value taken by simple averaging (macro average) between per class F1 values? or by weighted averaging (weighted average) of per class F1 values?
  2.  k=10 is somewhat 1:3:1 like class distribution. Is your model performs equally better on all classes or performs better for stationary class more?
  3.  Did you do any class weighting for training on k=10? If not can I know the characteristics of your model which handles the class skewness?

**Comment by PraAnj** [comment]
Thank you your prompt response. It seems this dataset is easy as achieving above 80% is really tough for this problem on other datasets. Anyways how did you get the LSE dataset, is it available for public/ or on request? Or you bought data from LSE? Seems worth having a training on that.

**Comment by zcakhaa** [comment]
We have access to Reuters and obtained the data there. You can try lobster order book data and purchase from them.


________________________________

**Comment by PraAnj** [comment]
Thank you, I will try that option.
